gsl/span

```diff
@@ -670,6 +670,19 @@ span<typename Ptr::element_type>
 make_span(Ptr& cont)
 { return span<typename Ptr::element_type>(cont); }
 
+template <class T, class = std::enable_if_t<std::is_pod<T>::value>>
```

Comment:
I can drop the restrictions, but my concern is that someone who reads 

    std::vector<int> data{...};
    auto raw_bytes = gsl::as_bytes(data);

will assume that `raw_bytes` is a span over the content of the vector and not over the private member variables. Especially as I don't believe there are a lot of cases, where you should / want to access the raw memory occupied by a complex object.